server/pkg/provision/auth.go

```diff
@@ -43,3 +61,120 @@ func emailSettings(ctx context.Context, s store.Storer) error {
 		})
 	})
 }
+
+// addAuthSuperUsers assigns BYPASS roles to users from AUTH_PROVISION_SUPER_USER value
+// When in Production, Corteza should stop and report an error.
+func addAuthSuperUsers(ctx context.Context, log *zap.Logger, s store.Storer) (err error) {
+	var (
+		val, has = os.LookupEnv("AUTH_PROVISION_SUPER_USER")
+		envOpt   = options.Environment()
+	)
+
+	if !has {
+		return
+	}
+
+	if envOpt.IsProduction() {
+		return errors.New(0, "Cannot provision super users in Production environment!")
+	}
+
+	users := strings.Split(val, ";")
+
+	for _, user := range users {
+		u := prepareUser(user)
+
+		// skip existing email
+		_, err = s.LookupUserByEmail(ctx, u.Email)
+		if err != store.ErrNotFound {
+			log.Warn(fmt.Sprintf("Email address already %s exists", u.Email))
+			continue
+		}
+
+		// skip existing handle
+		_, err = s.LookupUserByHandle(ctx, u.Handle)
+		if err != store.ErrNotFound {
+			log.Warn(fmt.Sprintf("Handle %s already exists", u.Handle))
+			continue
+		}
+
+		if err = store.CreateUser(ctx, s, u); err != nil {
+			return err
+		}
+
+		if err = setPassword(ctx, u.ID, u.Email, s); err != nil {
+			return err
+		}
+
+		log.Warn(fmt.Sprintf("User %#v created", u))
+
+		//assign the user a bypass role
+		for _, r := range internalAuth.BypassRoles() {
+			m := &types.RoleMember{UserID: u.ID, RoleID: r.ID}
+			if err = store.CreateRoleMember(ctx, s, m); err != nil {
+				return err
+			}
+		}
+	}
+
+	return
+}
+
+// prepareUser creates and fills a new user depending on the number of arguments provided
+func prepareUser(user string) *types.User {
+	u := &types.User{
+		ID:        nextID(),
+		CreatedAt: *now(),
+	}
+
+	usr := strings.Split(user, ",")
+	u.Email = usr[0]
+
+	if len(usr) > 2 {
+		u.Username = usr[1]
+		u.Name = usr[2]
+		u.Handle = createUserHandle(u)
+		return u
+	}
+
+	if len(usr) > 1 {
+		u.Username = usr[1]
+		u.Handle = createUserHandle(u)
+		return u
+	}
+
+	u.Handle = createUserHandle(u)
+	return u
+}
+
+func setPassword(ctx context.Context, userID uint64, password string, s store.Storer) (err error) {
```

Comment:
Did you check in the service code if there is any code you could reuse; perhaps a function call for this?